### PR TITLE
Refactoring.

### PR DESCRIPTION
**Pull request by AndreasAlbert**
First stab at #57. Any thoughts @clelange?

### CODE REVIEW

**Comment by AndreasAlbert** [comment]
Utility functions now live in `helpers` subspace. Everything to do with ROOT data formats moved to `RootUtils`.

**Comment by clelange** [comment]
Hi Andreas,

In general, this looks very good to me. I think it's good to abstract away the classes that should not be used by the user to some extent.
Could you reenable the pylint option for the > 1000 lines file?

It's a little bit inconsistent to use CamelCase for `RooUtils` and then all lower case `helpers`, use snake_case for both? Then we would have a convention that classes that should be used by the user are CamelCase and the other snake_case?

**Comment by AndreasAlbert** [comment]
Oh yeah, let's just use snake for both.

Guiding the user with the case convention used in the code is a bit too arcane for my taste.

**Comment by AndreasAlbert** [comment]
To be more precise:
CamelCase for class names, snake for everything else (which is what we did so far, I think)

**Comment by clelange** [comment]
I think there's only one error left:
```
hepdata_lib/__init__.py:258:12: E0602: Undefined variable 'execute_command' (undefined-variable)
```

**Comment by AndreasAlbert** [comment]
* Fixed pylint errors
* Applied snake_case to RootUtils submodule -> root_utils
* Travis now runs pylint on all python sub files

File: hepdata_lib/__init__.py [hunk 1]
```diff
@@ -22,6 +22,8 @@
 import numpy as np
```
Comment (clelange) [inline hepdata_lib/__init__.py]: We actually don't need the ROOT and numpy imports here anymore, do we?

Comment (AndreasAlbert) [reply hepdata_lib/__init__.py]: Yes, removed.

**Comment by AndreasAlbert** [comment]
You are right about the backward compatibility. For now, I added an import to the main file as per your suggestion. Longer term, we should encourage users to switch their import statements.

**Comment by clelange** [comment]
I'll work on #60 first so that we have a working docker container and binder examples again. Will get to it this week.

**Comment by AndreasAlbert** [comment]
If #60 is not a code issue on our side, why does this PR need to wait for it?

**Pull request merged**